app/controllers/oauth_clients_controller.rb

```diff
@@ -0,0 +1,56 @@
+class OauthClientsController < ApplicationController
+  before_action :login_required
+  before_action :get_client_application, :only => [:show, :edit, :update, :destroy]
+
+  def index
+    @client_applications = @current_user.client_applications
+    @tokens = @current_user.tokens.where('oauth_tokens.invalidated_at is null and oauth_tokens.authorized_at is not null')
+  end
+
+  def new
+    @client_application = ClientApplication.new
+  end
+
+  def create
+    @client_application = @current_user.client_applications.build(user_params)
+    if @client_application.save
+      flash[:notice] = "Registered the information successfully"
+      redirect_to :action => "show", :id => @client_application.id
+    else
+      render :action => "new"
+    end
+  end
+
+  def user_params
+      params.require(:client_application).permit(:name, :callback_url, :support_url, :url)
```

Comment:
The style's a bit strange on a lot of this code – e.g. here, where the indent doesn't match the rest of the file (4 spaces in from the method, but the method is 2 spaces in from the root), or like where we're using arrow syntax for hashes (`:action => "show"`) and strings for action values – basically all our other controllers use e.g. `render :new` syntax, which is implicitly the action, and hashes are done as e.g. `id: @client_application.id`.